src/urlBuilder.js

```diff
@@ -0,0 +1,58 @@
+const CONSTANTS = require('./constants');
+const { getMesh } = require('./lib/devConsole');
+
+const {
+	MULTITENANT_GRAPHQL_SERVER_BASE_URL,
+	EDGE_MESH_BASE_URL,
+	EDGE_MESH_SANDBOX_BASE_URL,
+} = CONSTANTS;
+
+/**
+ * Build the mesh url for the multitenant mesh.
+ *
+ * Gets the mesh details to checks for a custom domain in the case of a TI mesh.
+ * @param imsOrgId
+ * @param projectId
+ * @param workspaceId
+ * @param workspaceName
+ * @param meshId
+ * @param apiKey
+ * @returns {Promise<string>}
+ */
+async function buildMeshUrl(imsOrgId, projectId, workspaceId, workspaceName, meshId, apiKey) {
+	const { meshURL: customBaseUrl } = await getMesh(
+		imsOrgId,
+		projectId,
+		workspaceId,
+		workspaceName,
+		meshId,
+	);
+
+	return customBaseUrl
+		? `${customBaseUrl}/${meshId}/graphql`
+		: `${MULTITENANT_GRAPHQL_SERVER_BASE_URL}/${meshId}/graphql${
+				apiKey ? `?api_key=${apiKey}` : ''
+		  }`;
+}
+
+/**
+ * Builds the mesh url for the edge mesh.
+ *
+ * Uses the url for the appropriate Cloudflare namespace based on the console workspace name.
+ * @param meshId
+ * @param workspaceName
+ * @returns {string}
+ */
+function buildEdgeMeshUrl(meshId, workspaceName) {
+	let baseUrl;
+
+	if (EDGE_MESH_BASE_URL.includes('stage')) {
```

Comment:
I mentioned this on @kmaschi PR as well. This is a little tricky because cloudflare PROD has Prod and Sandbox, but you also have a stage CLI environment where prod and nonprod workspaces go to stage.

Reply:
I figured this constant would be safe to inspect since it will always be present, and its value will be representative of the env it was derived in. Thus, if the base url contains the word "stage", then it is the _ONLY_ url.

Do you have a particular suggestion?